packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -161,9 +164,25 @@ static int32_t getUniqueId()
 
 id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, std::shared_ptr<CallInvoker> jsInvoker)
 {
-  if (value.isUndefined() || value.isNull()) {
+  return convertJSIValueToObjCObject(runtime, value, jsInvoker, NO);
+}
+
+id convertJSIValueToObjCObject(
+    jsi::Runtime &runtime,
+    const jsi::Value &value,
+    std::shared_ptr<CallInvoker> jsInvoker,
+    BOOL useNSNull)
+{
+  if (value.isUndefined() || (value.isNull() && !useNSNull)) {
     return nil;
   }
+  if (value.isNull() && useNSNull) {
+    return [NSNull null];
+  }
+
+  if (value.isNull()) {
+    return [NSNull null];
+  }
```

Comment:
above `if` handles:

1- "undefined" and
2- "null && not using useNSNull"

These two cases handle:

1- "null && using NSNull" - which is partially redundant because it must be using NSNull, since !useNSNull was handled above, and
2- "null" - which would be either the useNSNull or !useNSNull case above, which is also redundant I think

Suggest collapsing them to one case, the one not already handled above

```suggestion
  if (value.isNull()) {
    return [NSNull null];
  }
```